lib/src/modules/apply.cc

```diff
@@ -225,43 +358,55 @@ exit exit_code
 
     Util::CommandObject Apply::buildCommandObject(const ActionRequest& request)
     {
+        // Shared
         auto params = request.params();
+        params.set<std::string>("ca", ca_);
+        params.set<std::string>("crt", crt_);
+        params.set<std::string>("key", key_);
+        params.set<std::string>("crl", crl_);
+        params.set<std::string>("proxy", proxy_);
+
         const fs::path& results_dir { request.resultsDir() };
-        // Ensure the ruby shim is in the cache dir.
-        // TODO: remove this when we figure out how to manage ruby shim
-        const std::string ruby_shim_cache_dir = "apply_ruby_shim";
-        const std::string ruby_shim_script_name = "apply_ruby_shim.rb";
+
+        const std::string ruby_shim_cache_dir = "apply_ruby_shims";
         const fs::path& cache_dir = module_cache_dir_->createCacheDir(ruby_shim_cache_dir);
-        auto ruby_shim_path = cache_dir / ruby_shim_script_name;
-        if (!fs::exists(ruby_shim_path)) {
-          lth_file::atomic_write_to_file(RUBY_SHIM, ruby_shim_path.string(), NIX_DOWNLOADED_FILE_PERMS, std::ios::binary);
+
+        std::string ruby_shim_script_name;
+        fs::path apply_ruby_shim_path;
+        std::string plugin_cache_name;
+
+        if (params.includes("catalog")) {
```

Comment:
We should see if there's a more robust way to decide between the two actions.

Reply:
https://github.com/puppetlabs/pxp-agent/blob/master/lib/inc/pxp-agent/action_request.hpp#L35 yeah i meant to comment there must be a better way. I think ^^ should do it.

Reply:
Updated to use that.